src/ucs/async/thread.h

```diff
@@ -9,13 +9,52 @@
 
 #include <ucs/type/spinlock.h>
 #include <ucs/sys/checker.h>
+#include <ucs/debug/assert.h>
+
+
+typedef struct ucs_async_thread_mutex {
```

Comment:
IMO it's overkill to implement own mutex just for sake of assertion
we can check lock.__owner field, or just not check at all, and only the "spinlock" mode will have the assertion checks

Reply:
fixed as discussed